src/basics.md

```diff
@@ -270,6 +271,54 @@ fn run() -> Result<()> {
 # quick_main!(run);
 ```
 
+[ex-regex-filter-log]: #ex-regex-filter-log
+<a name="ex-regex-filter-log"></a>
+## Filter a log file by matching multiple regular expressions
+
+[![regex-badge]][regex] [![cat-text-processing-badge]][cat-text-processing]
+
+Reads a file named `application.log` and only outputs the lines 
+containing “warning” or “error” (case insensitive). A [`regex::RegexSet`] is built
+with [`regex::RegexSetBuilder`] using the builder pattern.
+
+```rust
+# #[macro_use]
+# extern crate error_chain;
+extern crate regex;
+
+# use std::fs::File;
+# use std::io::{BufReader, BufRead};
```

Comment:
I would not hide these use statements. We generally hide only the error boilerplate or support code that is far from the point.